lxd/storage/backend_lxd.go

```diff
@@ -3357,6 +3397,77 @@ func (b *lxdBackend) EnsureImage(fingerprint string, op *operations.Operation) e
 	return nil
 }
 
+// shouldUseOptimizedImage determines if an optimized image should be used based on the provided volume config.
+// It returns true if the volume config aligns with the pool's default configuration, and an optimized image does
+// not exist or also matches the pool's default confgiuration.
+func (b *lxdBackend) shouldUseOptimizedImage(fingerprint string, volConfig map[string]string, op *operations.Operation) (bool, error) {
+	err := b.isStatusReady()
+	if err != nil {
+		return false, err
+	}
+
+	// Skip if the driver doesn't support optimized image volumes.
+	if !b.driver.Info().OptimizedImages {
+		return false, nil
+	}
+
+	// Load image info from database.
+	_, image, err := b.state.DB.Cluster.GetImageFromAnyProject(fingerprint)
+	if err != nil {
+		return false, err
+	}
+
+	// Derive content type from image type. Image types are not the same as instance types, so don't use
+	// instance type constants for comparison.
+	contentType := drivers.ContentTypeFS
+	if image.Type == "virtual-machine" {
+		contentType = drivers.ContentTypeBlock
+	}
+
+	// Create the image volume with the provided volume config.
+	newImgVol := b.GetVolume(drivers.VolumeTypeImage, contentType, fingerprint, volConfig)
+	err = b.Driver().FillVolumeConfig(newImgVol)
+	if err != nil {
+		return false, err
+	}
+
+	// Create the image volume with pool's default settings.
+	imgVol := b.GetVolume(drivers.VolumeTypeImage, contentType, fingerprint, nil)
+	err = b.Driver().FillVolumeConfig(imgVol)
+	if err != nil {
+		return false, err
+	}
+
+	// If the new volume's config doesn't match the pool's default configuration, don't use an optimized image.
+	if !volumeConfigsMatch(newImgVol, imgVol) {
```

Comment:
No need for this separate function, as there is only 1 call site, better to keep the logic altogether I think.

Reply:
ah no sorry I missed the 2nd call site.